src/main/java/se/leddy231/playertrading/mixin/BarrelEntityMixin.java

```diff
@@ -93,31 +100,52 @@ public IAugmentedBarrelEntity getNeighbourByType(BarrelType type) {
         return null;
     }
 
+    /**
+     * "Activate" will attempt to create
+     * a shop if all requirements are
+     * met.
+     * Requirements:
+     * 1. The shop doesn't already have an owner
+     * 2. The shop is valid and has all its parts intact
+     */
     @Override
-    public void activate(PlayerEntity player, BarrelType signType) {
+    public void activate(PlayerEntity player, BarrelType signType, String shopName) {
+
+        // TODO
+        /* Not fully implemented yet
+        if (!completeSetup(signType)) {
+            Utils.sendMessage(player, (LiteralText) new LiteralText("Incomplete setup, shop must be connected to a stock and an output").formatted(Formatting.RED));
+            return;
+        }*/
+
         if (owner != null) {
             if (owner.equals(player.getUuid())) {
                 playerTroubleshoot(player);
             } else {
-                Utils.sendMessage(player, "Someone already owns this");
+                Utils.sendMessage(player, (LiteralText) new LiteralText("Someone already owns this").formatted(Formatting.RED));
             }
         } else {
-            create(player, signType);
+            create(player, signType, shopName);
         }
     }
 
-    public void create(PlayerEntity player, BarrelType newType) {
+    /**
+     * "Create" will instantiate a shop
+     * and set its corresponding fields
+     */
+    public void create(PlayerEntity player, BarrelType newType, String shopName) {
         owner = player.getUuid();
         type = newType;
+        this.shopName = shopName;
 
         BarrelBlockEntity entity = (BarrelBlockEntity) (Object) this;
         entity.markDirty();
 
-        Utils.sendMessage(player, newType.typeName() + " barrel created");
+        Utils.sendMessage(player, (LiteralText) new LiteralText(newType.typeName() + " barrel created").formatted(Formatting.GREEN));
     }
 
     public void playerTroubleshoot(PlayerEntity player) {
-        Utils.sendMessage(player, type.typeName() + " barrel owned by you");
+        Utils.sendMessage(player, (LiteralText) new LiteralText(type.typeName() + " barrel already owned by you").formatted(Formatting.RED));
```

Comment:
This should probably be green, as it is shown when the player tries to troubleshoot the shop by right-clicking the sign again